### ISSUE TITLE
Don't save docs that are already in the right state

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#4688]
Following on from #4109 where we don't update the message state if it's the same as the current message state to stop the history getting too long, we should also not persist the doc if none of the messages actually got updated. Doing so causes many unnecessary database writes with only a _rev change, which significantly impacts the server.

**Steps to reproduce**:
- File a report for which sentinel generates scheduled tasks.
- Use the `/api/sms` API to POST a status update which sets the state of a message to `received-by-gateway`.
- Verify that the message state in the db is updated and record the _rev of the doc at this point.
- Use the `/api/sms` API to set the state of the same message to `received-by-gateway` again.

**What should happen**:
 - The doc is not updated, which can be verified by the _rev remaining the same.

**What actually happens**:
 - The doc is updated. The _rev is the only field that changes.

**Environment**:

- Instance: zazic-zimbabwe.app.medicmobile.org
- Browser: N/A
- Client platform: N/A
- App: gateway, api
- Version: 2.15.0

**Comment by garethbowen** [comment]
Merged into `2.16.x` and forward ported to `master`.

**Comment by garethbowen** [comment]
@ngaruko 

> a) is the message referred to in step 2 one of those generated in step one?

Yes

> b) what the 'status update which sets the state of a message to received-by-gateway.' seems not to clear from the documentation where there are 4 statuses : PENDING|SENT|DELIVERED|FAILED.

Use "DELIVERED", sorry for the confusion.

**Comment by garethbowen** [comment]
I've just ATed this and it passed. Here's what I did...

1. I found a message UUID from the `medic/tasks_messages` view. The one I picked was `0013639e-830a-44d3-8b6a-5e2a0f22952d` in doc `5a5298b0a308146efa4ab51d74b0781e`.
2. I checked and recorded the rev started with 3
3. I used this curl command documented in the medic-gateway docs: `curl https://admin:pass@beta-old.dev.medicmobile.org/api/sms -H "Accept: application/json" -H "Accept-Charset: utf-8" -H "Cache-Control: no-cache" -H "Content-Type: application/json" --data '{"messages": [], "updates": [{"id": "0013639e-830a-44d3-8b6a-5e2a0f22952d", "status": "DELIVERED"}]}'`
4. I checked the doc again and saw the rev started with 4 and the message status had been updated correctly.
5. I repeated the curl from above
6. I checked the doc again and the rev had not been changed and the message state was still correct.
